yapapi/executor/__init__.py

```
@@ -368,42 +368,30 @@ async def accept_payment_for_agreement(agreement_id: str, *, partial: bool = Fal
             )
 
         async def find_offers() -> None:
-            async def reject_proposal(proposal, reason):
-                try:
-                    await proposal.reject(reason=reason)
-                    emit(events.ProposalRejected(prop_id=proposal.id, reason=reason))
-                except Exception:
-                    emit(
-                        events.ProposalFailed(
-                            prop_id=proposal.id, exc_info=sys.exc_info()  # type: ignore
-                        )
-                    )
+            """Subscribe to offers and process them continuously."""
 
-            async def respond_to_proposal(proposal, builder):
-                try:
-                    await proposal.respond(builder.properties, builder.constraints)
-                    emit(events.ProposalResponded(prop_id=proposal.id))
-                except Exception:
-                    emit(
-                        events.ProposalFailed(
-                            prop_id=proposal.id, exc_info=sys.exc_info()  # type: ignore
-                        )
-                    )
+            async def reject_proposal(proposal, reason):
+                await proposal.reject(reason=reason)
+                emit(events.ProposalRejected(prop_id=proposal.id, reason=reason))
```

It's safer not to catch any exceptions here and let them bubble up